addons/test_mail/tests/test_message_composer.py

```diff
@@ -307,6 +381,90 @@ def test_composer_w_template_mass_mailing(self):
+            'email_from': 'laurie.poiret@example.com',
+            'name': 'Test2',
+        })
+        existing_partners = self.env['res.partner'].search([])
+
+        self.email_template.write({
+            'email_cc': '"Raoul" <test.cc.1@example.com>, invalid',
+            'email_from': '${user.email_formatted | safe}',
+            'email_to': '"Micheline, l\'Immense" <test.to.1@example.com>, test.to.2@example.com, wrong',
+            'partner_to': '%s,%s,%s' % (self.partner_1.id, self.partner_2.id, 0)
+        })
+        self.user_employee.write({'email': 'email.from.1@test.example.com, email.from.2@test.example.com'})
+        self.partner_1.write({'email': '"Valid Formatted" <valid.lelitre@agrolait.com>'})
+        self.partner_2.write({'email': 'valid.other.1@agrolait.com, valid.other.cc@agrolait.com'})
+
+        composer = self.env['mail.compose.message'].with_user(self.user_employee).with_context({
+            'active_ids': [self.test_record.id, test_record_2.id],
+            'default_composition_mode': 'mass_mail',
+            'default_model': 'mail.test.simple',
+            'default_notify': False,
+            'default_res_id': self.test_record.id,
+            'default_template_id': self.email_template.id,
+        }).create({})
+        composer.write(
+            composer.onchange_template_id(
+                self.email_template.id, 'mass_mail', 'mail.test.simple', self.test_record.id
+            )['value']
+        )
+        composer.send_mail()
+
+        # find partners created during sending (as emails are transformed into partners)
+        new_partners = self.env['res.partner'].search([]).search([('id', 'not in', existing_partners.ids)])
+        self.assertEqual(
+            sorted(new_partners.mapped('email')),
+            sorted(['test.cc.1@example.com', 'test.to.1@example.com', 'test.to.2@example.com']),
+            'Mail: created partners for valid emails (wrong / invalid not taken into account)'
+        )
+        self.assertEqual(
+            sorted(new_partners.mapped('name')),
+            sorted(['test.cc.1@example.com', 'test.to.1@example.com', 'test.to.2@example.com']),
+            'Mail: currently setting name = email, not taking into account formatted emails'
+        )
+
+        # hack to use assertEmails
+        _mails_record1 = [mail for mail in self._mails if '%s-%s' % (self.test_record.id, self.test_record._name) in mail['message_id']]
+        _mails_record2 = [mail for mail in self._mails if '%s-%s' % (test_record_2.id, test_record_2._name) in mail['message_id']]
+        for record, mails in zip(
+                self.test_record + test_record_2,
+                [_mails_record1, _mails_record2]
+            ):
+            # specific for customer with formatted email
+            partner1_email = next(
+                mail for mail in mails
+                if mail['email_to'] == [tools.formataddr((self.partner_1.name, 'valid.lelitre@agrolait.com'))]
+            )
```

Comment:
(Sorry for the spam, just noticed this also)

Your `next` usages in your tests don't use a default parameter.
Which means your tests will fail with a StopIteration exception instead of failing nicely on your assertTrue here below.
(There are multiple usages through the PR, I let you check all of them).